edb/server/server.py

```diff
@@ -371,13 +371,46 @@ def cancel_pgext_connection(self, pid, secret):
     def monitor_fs(
         self,
         path: str | pathlib.Path,
-        cb: Callable[[str, int], None],
+        cb: Callable[[], None],
     ) -> Callable[[], None]:
         if not self._use_monitor_fs:
             return lambda: None
 
+        handle = None
+        parent_dir = pathlib.Path(path).parent
+
+        def watch_dir(file_modified, _event):
+            nonlocal handle
+            if parent_dir / os.fsdecode(file_modified) == pathlib.Path(path):
+                try:
+                    new_handle = self.__loop._monitor_fs(  # type: ignore
+                        str(path), callback)
+                except FileNotFoundError:
+                    pass
+                else:
+                    finalizer()
+                    handle = new_handle
+                    self._file_watch_handles.append(handle)
+
+        def callback(_file_modified, event):
+            nonlocal handle
+            if event == 2:  # CHANGE
+                cb()
+            else:  # RENAME, RENAME_CHANGE
```

Comment:
Test for these events explicitly. Otherwise you might react wrongly to a non-rename event.